packaging/nuget/packaging.py

```diff
@@ -523,74 +298,80 @@ def xlayout (self):
             if 'toolset' not in a.info:
                 a.info['toolset'] = 'v120'
 
-        nuget_layout = {
-            # Build
-            'build/native/lib/${plat}/${arch}/${variant}/${toolset}/': 'static',
-            'build/native/include/librdkafka/': 'include',
-
-            # Runtime
-            'runtimes/${plat}-${arch}/native/': 'dynamic',
-
-            # All
-            'content/docs/': 'doc'
-        }
-
-        layout = self.layout(nuget_layout)
-
-        errors = 0
-        print(' %s layout:' % self)
-        for path, afs in layout.items():
-            print('  %s provided by:' % path)
-            for a, f in afs:
-                print('    %s from artifact %s (and %d more)' % (f, a.fname, len(afs)-1))
-                break
-            if len(afs) == 0:
-                print('     ERROR: no artifacts found')
-                errors += 1
-        print('')
-
-        if errors > 0:
-            raise Exception('Layout not satisfied by collected artifacts: %d missing' % errors)
-
-        return layout
-
-
-
+        mappings = [
+            [{'arch': 'x64', 'plat': 'linux', 'fname_startswith': 'librdkafka.tar.gz'}, './include/librdkafka/rdkafka.h', 'build/native/include/librdkafka/rdkafka.h'],
+            [{'arch': 'x64', 'plat': 'linux', 'fname_startswith': 'librdkafka.tar.gz'}, './include/librdkafka/rdkafkacpp.h', 'build/native/include/librdkafka/rdkafkacpp.h'],
+
+            [{'arch': 'x64', 'plat': 'osx', 'fname_startswith': 'librdkafka.tar.gz'}, './lib/librdkafka.dylib', 'runtimes/osx-x64/native/librdkafka.dylib'],
+            [{'arch': 'x64', 'plat': 'linux', 'fname_startswith': 'librdkafka-debian9.tgz'}, './lib/librdkafka.so.1', 'runtimes/linux-x64/native/debian9-librdkafka.so'],
+            [{'arch': 'x64', 'plat': 'linux', 'fname_startswith': 'librdkafka.tar.gz'}, './lib/librdkafka.so.1', 'runtimes/linux-x64/native/librdkafka.so'],
+
+            [{'arch': 'x64', 'plat': 'win7', 'fname_startswith': 'msvcr120.zip'}, 'msvcr120.dll', 'runtimes/win7-x64/native/msvcr120.dll'],
+            [{'arch': 'x64', 'plat': 'win7', 'fname_startswith': 'librdkafka.redist'}, 'build/native/bin/v120/x64/Release/librdkafka.dll', 'runtimes/win7-x64/native/librdkafka.dll'],
```

Comment:
It would be good with a small comment what filename it expects when only specifying a prefix.
E.g.,
`# from librdkafka.redist.VER.nupkg`

For the one below saying just `librdkafka` in particular.

Alternatively if a glob name was used instead.

Reply:
yep, good idea. it's unfortunate the naming of the artifacts makes differentiating them tricky.